File: 6_DES_COMP/Des_Comp_01.py
'''
Dado un entero N donde cada digito es diferente de 0, se pide ordenar el numero de mayor
a menor.
'''
#1 Y 5 EN REVISION, NO ESTA ACABADO
import math
N=int(input("N: "))
while N<=0:
    print("N debe ser mayor o igual a 0")
    N=int(input("N: "))
dm=9; dM=0
cd=int(math.log10(N)+1)
po=1; nn=0
while N>0:
    d=N%10
    N=N//10
    temp=N
    cdtemp=int(math.log10(N)+1)
    while temp>0:
        dtemp=temp//10**(cd-1)
        temp=temp%10**(cd-1)
        cd=cd-1
        if d<dtemp:
            d
print(nn)

File: 6_DES_COMP/Des_Comp_01.java
/*
 * Dado un entero N donde cada digito es diferente de 0, se pide ordenar el numero de mayor a menor.
 */
import java.util.Scanner;

public class Des_Comp_01 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("N: ");
        int N = scanner.nextInt();

        while (N <= 0) {
            System.out.println("N debe ser mayor o igual a 0");
            System.out.print("N: ");
            N = scanner.nextInt();
        }

        String numeroStr = Integer.toString(N);
        char[] digitos = numeroStr.toCharArray();
        
        // Ordenar los dígitos de mayor a menor
        for (int i = 0; i < digitos.length - 1; i++) {
            for (int j = i + 1; j < digitos.length; j++) {
                if (digitos[i] < digitos[j]) {
                    char temp = digitos[i];
                    digitos[i] = digitos[j];
                    digitos[j] = temp;
                }
            }
        }
        
        String numeroOrdenado = new String(digitos);
        System.out.println("Número ordenado: " + numeroOrdenado);
        
        scanner.close();
    }
}